### PR TITLE
LateNight tweaks

### PR DESCRIPTION
**Pull request by ronso0**
- [x] more intuitive icons for beatgrid controls, hotcue shift buttons, looping & beatjumping:New Loop, beatjump & beatgrid icons:
image
- [x] add track color to decks
- [x] add 16 samplers option
image
- [x] compact controls in skin settings for number controls (count of decks, Fx units & samplers)

* ..anything else?

### CODE REVIEW

**Comment by JoergAtGithub** [comment]
The buttons are much better to understand now! What do you think about changeing the beatgrid shift buttons from 
grafik to 
grafik
IMHO this makes it more clear, that the whole beatgrid is affected, because the right edge of the arrow is not on a beat.

**Comment by JoergAtGithub** [comment]
On a second thought, would it be more clear, to have the base edge of the arrows always on the object to move e.g. 
grafik
instead of 
grafik

**Comment by JoergAtGithub** [comment]
And maybe this 
grafik instead of 
grafik

**Comment by ywwg** [comment]
> On a second thought, would it be more clear, to have the base edge of the arrows always on the object to move e.g.

+1 to both of these suggestions

**Comment by JoergAtGithub** [comment]
What do you think about the this sketch:
grafik
I changed the following:
-Move Cue Left/Right is now horizontal below Move Beatgrid Left/Right
-The Adjust BPM Up/Down buttons are now vertical (as the pitch fader)
-I added a third beat bar to the Adjust BPM buttons. Now all buttons, that affect the whole beatgrid have 3 beat bars

**Comment by ronso0** [comment]
hotcue_shift buttons are optional, it would be confusing of other buttons change their position when shift buttons are shown.
I think beats_slower/_faster is good now, the third bar doesn't add information IMO.

**Comment by ronso0** [comment]
I changed the looping & beatjump icons so they differ from the beatgrid icons.
What do you think?
image

**Comment by Holzhaus** [comment]
I like the new loop in/loop out buttons. I'm unsure about the beatjump buttons, because they look a bit like fast forward/rewind, not beatjump. On the other hand, the old buttons looked like skip forward/back buttons and I have no idea what a "beatjump icon" could look like, so I guess the fast forward/rewind symbols fit better than the alternative.

While you're at it, could you have a look at the remaining skin errors?
```
Warning [Main]: Property "highlight" was not defined for widget "Spinbox_embedded" of type WBeatSpinBox (parameter: QVariant(double, 0) )
[...]
Warning [Main]: Property "highlight" was not defined for widget "Spinbox_embedded" of type WBeatSpinBox (parameter: QVariant(double, 0) )
Warning [Main]: WaveformSignalColors::fallBackDefaultColor - Skin does not provide valid signal colors, using default color...
```

**Comment by JoergAtGithub** [comment]
> I like the new loop in/loop out buttons. I'm unsure about the beatjump buttons, because they look a bit like fast forward/rewind, not beatjump.
I aggree to both, what do you think about this: 
grafik

**Comment by ronso0** [comment]
adapting the Shade icons :) Sure!

**Comment by ronso0** [comment]
I pushed the icons back a bit, I'm about to finish adding 16 samplers. now it's just about ironing out some glitches in the sin settings.

**Comment by ronso0** [comment]
Okay, 16 samplers are in place and toggling works nicely.

I added some extra in-line toggles for deck count, hotcues, Fx units and sampler count.
Skin settings can are a bit more compact (now / before)
Screenshot_2020-10-10_10-27-56


New Loop, beatjump & beatgrid icons:
image

**Comment by JoergAtGithub** [comment]
Did you consider to use green as color for the active loop button, instead of orange? Than loops in the waveforms and the loop buttons would have the same color:
grafik
IMHO orange looks better, while green would result in functional consistency.

**Comment by ronso0** [comment]
Hm, actually I decided to have all toggles in the transport row orange to keep static toggles consitent (in contrast to colored hotcues) and to lower the confusion.
the other way around: what about painting the loop orange? Did you try that?

**Comment by ronso0** [comment]
> While you're at it, could you have a look at the remaining skin errors?
> 
> ```
> Warning [Main]: Property "highlight" was not defined for widget "Spinbox_embedded" of type WBeatSpinBox (parameter: QVariant(double, 0) )
> [...]
> Warning [Main]: Property "highlight" was not defined for widget "Spinbox_embedded" of type WBeatSpinBox (parameter: QVariant(double, 0) )

The use of `highlight` property works, therefore the warning is pointless. I have WIP branch to add highlight to the base widget.

> Warning [Main]: WaveformSignalColors::fallBackDefaultColor - Skin does not provide valid signal colors, using default color...
> ```

I'll check that!

**Comment by JoergAtGithub** [comment]
> the other way around: what about painting the loop range and the waveform marker orange? Did you try that?
grafik

**Comment by ronso0** [comment]
Nice! Never tried that and so I didn't notice the underlying waveform shines through that way.
Other's opinions on that? @Holzhaus @ywwg

**Comment by JoergAtGithub** [comment]
This is just painted, to get a visual impression

**Comment by ronso0** [comment]
hmkay..
[here](https://github.com/mixxxdj/mixxx/blob/master/res/skins/LateNight/skin.xml#L162)'s the place in case you wanna play a bit with it.

**Comment by ywwg** [comment]
I'm getting a cmake error:

```
CMake Error at CMakeLists.txt:1233 (add_executable):
  Target "mixxx-test" links to target "KeyFinder::KeyFinder" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
```

**Comment by ywwg** [comment]
(same error for other build targets)

**Comment by ronso0** [comment]
I don't see how this could be related to changes in this PR.
Does building 2.3 work for you?

**Comment by ronso0** [comment]
CI succeeds building

**Comment by Holzhaus** [comment]
@ywwg this happens because you built Mixxx with keyfinder support, then checked out this branch which has probably been forked off before keyfinder support has been added. Just delete your build directory and rebuild, then the issue is gone.

**Comment by Holzhaus** [comment]
@ronso0 is this ready to review and merge?

**Comment by ronso0** [comment]
Yes, I'm confident the entire samplers section works flawlessly now with the Singletons.
Please test!

**Comment by ywwg** [comment]
This looks good!  Maybe the default should be for the samplers to be collapsed -- they take up a lot of space expanded.

**Comment by ronso0** [comment]
> This looks good! Maybe the default should be for the samplers to be collapsed -- they take up a lot of space expanded.

Oh, my bad. I commited the test setup..
Default is 8 minimized samplers.

File: res/skins/LateNight/style_palemoon.qss [hunk 1]
```diff
@@ -2835,10 +2844,12 @@ WEffectSelector::item,
   WTrackMenu QMenu QCheckBox:focus,
   WTrackMenu QMenu QCheckBox:hover,
   #SkinSettingsButton[hover="true"],
-  #DeckSizeButton[hover="true"],
+  #SkinSettingsNumToggle[hover="true"],
+  #SkinSettingsNumToggleHeader[hover="true"],
   #SkinSettingsMixerToggle[hover="true"],
   #SkinSettingsLabelButton[hover="true"] {
-    background-color: #2c454f;
+    background-color: 2#c454f;
```
Comment (Holzhaus) [inline res/skins/LateNight/style_palemoon.qss:2851]:
```
Warning [Main]: QCssParser::parseHexColor: Unknown color name '#c454f'
```

**Review by Holzhaus — CHANGES_REQUESTED** [review]
Thank you! All looks good except for the small typo in the qss file (see below).

File: res/skins/LateNight/style_palemoon.qss (hunk 1 above)
Comment (ronso0) [reply res/skins/LateNight/style_palemoon.qss]: thanks, fixed!

**Review by Holzhaus — APPROVED** [review]
LGTM, thank you!

**Pull request merged**